File: project/app/src/main/java/org/owntracks/android/injection/modules/ServiceModules/BackgroundServiceModule.java
package org.owntracks.android.injection.modules.ServiceModules;

import android.app.Service;

import org.owntracks.android.injection.modules.BaseServiceModule;
import org.owntracks.android.injection.scopes.PerActivity;
import org.owntracks.android.services.BackgroundService;

import dagger.Binds;
import dagger.Module;

@Module(includes = BaseServiceModule.class)
public abstract class BackgroundServiceModule {

    @Binds
    @PerActivity
    abstract Service bindService(BackgroundService s);

}

File: project/app/src/main/java/org/owntracks/android/services/BackgroundServiceModule.kt
package org.owntracks.android.services

import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ServiceComponent

@InstallIn(ServiceComponent::class)
@Module
abstract class BackgroundServiceModule {
//    @Binds
//    @ServiceScoped
//    abstract fun bindServiceContext(service: Service): Context
}